pages/project-details/[id]/index.js

```diff
@@ -71,6 +76,38 @@ export default function ProjectDetails({
     setEditImage(false);
   }
 
+  function addNewDimensionProject(id, newDimension) {
+    setProjects(
+      projects.map((project) => {
+        if (project.id === id) {
+          return {
+            ...project,
+            dimensions: [...project.dimensions, newDimension],
+          };
+        } else {
+          return project;
+        }
+      })
+    );
+  }
+
+  function handleDimensionDeleteProjects(id) {
+    setProjects(
+      projects.map((project) => {
+        if (project.id === currentProject.id) {
+          return {
+            ...project,
+            dimensions: project.dimensions.filter(
+              (dimension) => dimension.id !== id
+            ),
+          };
+        } else {
+          return project;
+        }
+      })
+    );
+  }
```

Comment:
Maybe I'm wrong but isn't the name dimension a bit wrong? Isn't it Material? Because it's the `Materiallist`?

Reply:
yes you right, i thought about adding new dimensions which is part of that but its all about the material so i changed that 🙂